src/main/java/com/minecolonies/coremod/commands/GetColonyAndCitizen.java

```diff
@@ -0,0 +1,196 @@
+package com.minecolonies.coremod.commands;
+
+import com.minecolonies.coremod.colony.ColonyManager;
+import net.minecraft.command.ICommandSender;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.UUID;
+
+/**
+ * Get the Colony ID and Citizen ID out of a command
+ */
+public abstract class GetColonyAndCitizen
+{
+
+    private static final String ONLY_NUMBERS       = "Please only use numbers for the %s ID!";
+    private static final String TOO_MANY_ARGUMENTS = "Too many arguments!";
+    private static final String UNKNOWN_ERROR      = "Unknown Error!";
+    private static final String NOT_FOUND          = "%s not found";
+    private static final String NO_COLONY          = "You haven't got a colony!";
+
+
+    public static int colonyId;
+    public static int citizenId;
+
+    public static int getColonyId(@NotNull final ICommandSender sender, @NotNull final String... args)
+    {
+        final UUID mayorID = sender.getCommandSenderEntity().getUniqueID();
+        if (args.length == 2 || args.length == 4)
+        {
+            try
+            {
+                colonyId = Integer.parseInt(args[0]);
+            }
+            catch (NumberFormatException e)
+            {
+                colonyId = -1;
+            }
+        }
+        else if (args.length == 1 || args.length == 3)
+        {
+            try
+            {
+                colonyId = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), mayorID).getID();
+            }
+            catch (NullPointerException e)
+            {
+                colonyId = -5;
+            }
+        }
+        else if (args.length >= 5)
+        {
+            colonyId = -2;
+        }
+        else
+        {
+            colonyId = -3;
+        }
+        if (colonyId >= 0)
+        {
+            if (ColonyManager.getColony(colonyId) == null)
+            {
+                colonyId = -4;
+            }
+        }
+
+
+        return colonyId;
+    }
+
+    public static int getCitizenId(@NotNull final int colonyId, @NotNull final String... args)
+    {
+        String citizenName;
+        citizenId = -4;
+        if (args.length == 2)
+        {
+            try
+            {
+                citizenId = Integer.parseInt(args[1]);
+            }
+            catch (NumberFormatException e)
+            {
+                citizenId = -1;
+            }
+        }
+        else if (args.length == 1)
+        {
+            try
+            {
+                citizenId = Integer.parseInt(args[0]);
+            }
+            catch (NumberFormatException e)
+            {
+                citizenId = -1;
+            }
+        }
+        else if (args.length == 3 && colonyId >= 0)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)